api/controllers/zetter.ts

```diff
@@ -1,296 +1,306 @@
-import tweets from '../data/tweets';
-import users from '../data/users';
-import followers from '../data/followers';
-import favorities from '../data/favorities';
 import { sign } from 'jsonwebtoken';
 import { createHash } from 'crypto';
 import { Request, Response } from 'express';
-import { FavoriteType, FollowerType, ProfileType, TweetType } from '../utils/types';
+import { prisma } from '../utils/prisma';
+import { Tweet, User, Favorite } from '@prisma/client';
 
-export const getTweets = (req: Request, res: Response) => {
+export const getTweets = async (req: Request, res: Response) => {
     const limit: number = 10;
-    const followingUsers: Array<FollowerType> = followers.filter((follower) => follower.from === req.user.id);
-    const followingUsersAndMyIds: Array<number> = followingUsers.map((obj) => obj.to);
-    let followingUsersTweetsList: Array<TweetType> = [];
-
-    followingUsersAndMyIds.push(req.user.id);
-    tweets.forEach((tweet) => {
-        if (followingUsersAndMyIds.includes(tweet.createdBy)) {
-            followingUsersTweetsList.push(tweet);
-        }
+    const followings = await prisma.user.findMany({
+        where: {
+            followedBy: {
+                some: { id: req.user.id },
+            },
+        },
     });
-
-    followingUsersTweetsList.sort(function (a: TweetType, b: TweetType) {
-        return Number(b.createdAt) - Number(a.createdAt);
+    const timelineTweetsList: Array<Tweet> = await prisma.tweet.findMany({
+        where: {
+            createdBy: {
+                in: [req.user.id, ...followings.map((following) => following.id)],
+            },
+        },
+        include: {
+            user: true,
+            favorities: true,
+            replyFrom: true,
+        },
+        orderBy: {
+            createdAt: 'desc',
+        },
+        take: limit,
     });
-
-    const result: Array<TweetType> = addInformationToTweet(followingUsersTweetsList, req);
-    res.json(result.slice(0, limit));
+    res.json(timelineTweetsList);
 };
 
-export const getSpecificUsersTweets = (req: Request, res: Response) => {
+export const getSpecificUsersTweets = async (req: Request, res: Response) => {
     const limit: number = 10;
-    const user: ProfileType | undefined = users.find(({ username }) => username === req.params.username);
-    const sortedTweets: Array<TweetType> = tweets.sort(function (a: TweetType, b: TweetType) {
-        if (Number(a.createdAt) - Number(b.createdAt)) {
-            return -1;
-        } else {
-            return 1;
-        }
+    const user: User | null = await prisma.user.findUnique({
```

Comment:
ここは逆に、userとtweetsを分けてとるんじゃなくて、一緒にとってきてもいいかも

```suggestion
    const user: User | null = await prisma.user.findUnique({
      where: { username: req.params.username },
      include: { 
          tweets: { 
              orderBy: {
                  { createdBy: 'desc' }
              },
              take: limit,
              include: {
                  user: true,
                  favorities: true,
                  replyFrom: true,
              }
          }
      }
    })
```